cloudtasks/pull_queue/main.go

```diff
@@ -0,0 +1,194 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"golang.org/x/net/context"
+
+	cloudtasks "cloud.google.com/go/cloudtasks/apiv2beta2"
+	duration "github.com/golang/protobuf/ptypes/duration"
+	taskspb "google.golang.org/genproto/googleapis/cloud/tasks/v2beta2"
+)
+
+// usage powers the help documentation when a help operation is used or the wrong number of arguments specified.
+func usage() {
+	fmt.Println("Usage of tasks_cli:")
+	fmt.Println()
+	fmt.Println("\t$> tasks create $PROJECT_ID $LOCATION_ID $QUEUE_ID")
+	fmt.Println("\t$> tasks pull $PROJECT_ID $LOCATION_ID $QUEUE_ID")
+	fmt.Println("\t$> tasks help")
+	fmt.Println("\nFor more information, see https://cloud.google.com/cloud-tasks/docs")
+}
+
+func main() {
```

Comment:
Do we need a `main` for this sample? The code is normally run by copying & pasting from cloud.google.com, so the recommendation is to not write a `main` unless it's part of the sample (recent guidance). See https://github.com/GoogleCloudPlatform/golang-samples/issues/530.

Reply:
That's a good question, and might fit better with the documentation. In this case, [nodejs](https://github.com/googleapis/nodejs-tasks/tree/master/samples), [python](https://github.com/GoogleCloudPlatform/python-docs-samples/tree/master/tasks), and [Jave](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/tasks/src/main/java/com/example/Quickstart.java) have all been setup as a full CLI app from which individual functions are defined as snippets.

Should the golang sample go another route?

Reply:
Yes, I think so.

Reply:
* [x] Refactored from CLI to snippets